### PR TITLE
Add limit to share

### PR DESCRIPTION
**Pull request by loiczam**
Feature for this issue : 
https://github.com/photobooth-app/photobooth-app/issues/295

### CODE REVIEW

**Comment by loiczam** [comment]
Admin configuration :
admin

Frontend display error :
error

**Comment by loiczam** [comment]
Limit share/print by action

**Comment by mgineer85** [comment]
Hey, I am sorry, but I have hard time using the application config as a database.
There is the stats counter, a limiting implementation better relies on that.

**Comment by loiczam** [comment]
> Hey, I am sorry, but I have hard time using the application config as a database. There is the stats counter, a limiting implementation better relies on that.

Hi, 
I used config instead stats counter because I think a counter by action is a better solution. In a first version i used the stats counter.

File: photobooth/services/config/appconfig.py [hunk 1]
```diff
@@ -171,6 +171,47 @@ def get_schema(cls, schema_type: str = "default"):
     def reset_defaults(self):
         self.__dict__.update(__class__())
 
+    def update_field(self, field_path: str, value: Any):
```
[the remark this answers is not in the document]
Comment (loiczam) [reply photobooth/services/config/appconfig.py:174]: It's for update a field. thx for the tips I check that

**Comment by mgineer85** [comment]
Yes, counter divided is better. That’s part of the reason I did not want to implement. I don’t think it’s needed in an open software and quite effort to implement it right 😌

File: photobooth/services/config/appconfig.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (loiczam) [reply photobooth/services/config/appconfig.py]: I fix it

**Comment by loiczam** [comment]
> Yes, counter divided is better. That’s part of the reason I did not want to implement. I don’t think it’s needed in an open software and quite effort to implement it right 😌

@mgineer85 if we use the stats counter, we must reset everything if we want to reset just the limit counter of an action.
I think 

Although the stats counter is a great solution, I think it is not possible to use it for this feature in its current state.

**Comment by mgineer85** [comment]
Well, as we extend the functionality to add a share/print limitation, it is not about just implementing this function no matter what. If it needs updated underlying code to support the desired functionality properly the underlying code needs revisal also.

If the underlying change is to allow resetting the stats separately it should be added. It needs an rest api method and buttons in the frontend for the admin then.

**Comment by loiczam** [comment]
@mgineer85 
Hello,

I have change for using stats counter and I add an api to reset limites.

**Comment by mgineer85** [comment]
Hey, the linter is complaining about some items, see above in the details.
Could you have a look?

I tested also, I think it would be good if the limit is checked before the blocking time. This avoids user waiting and finally running after block time into the limit being disappointed.

Also now there is the additional api endpoint but it's not used in the frontend, yet. Do you really need a separate reset to the limit counter or is it fine to reset just all?

**Comment by loiczam** [comment]
> Hey, the linter is complaining about some items, see in the details. Could you have a look?
> 
> I tested also, I think it would be good if the limit is checked before the blocking time. This avoids user waiting and finally running after block time into the limit being disappointed.
> 
> Also now there is the additional api endpoint but it's not used in the frontend, yet. Do you really need a separate reset to the limit counter or is it fine to reset just all?

I change test order on my last commit.

To reset only the limits, I am looking into doing the development on the frontend.

**Pull request merged**